mitm.go

```diff
@@ -119,15 +120,20 @@ var (
 )
 
 func (p *Proxy) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	if p.Director == nil {
```

Comment:
This is also necessary for `ReverseProxy` so it makes sense to be at the beginning (it already happened, just moved it up).